common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -226,6 +226,14 @@ private static boolean isSymlink(File file) throws IOException {
    * The unit is also considered the default if the given string does not specify a unit.
    */
   public static long timeStringAs(String str, TimeUnit unit) {
+    return timeStringAs(str, unit, unit);
+  }
+
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a time count in the given unit.
+   * defaultUnit is used for string which have just number and no units mentioned
+   */
+  public static long timeStringAs(String str, TimeUnit unit, TimeUnit defaultUnit) {
```

Comment:
Without looking at the PR, +1 to Sean's comment. The "no unit" thing was always a backwards compatibility hack. I think it's time we drop support for that since it's confusing.

(I'd also vote for making all time configs return milliseconds but that's a much noisier change.)

Reply:
so i see two options
1. Handle all other time related configurations also to fall back as milliseconds by default when units are not mentioned
2. Or disallow any time related configuration to be configured without specifying unit

Please suggest which one i can proceed with.?

Reply:
Let's not disallow unit-less time, not here, though I think that's reasonable to consider separately for Spark 3. Here I think we should just focus on fixing the inconsistency in handling this one parameter, and make it consistent with how it's handled in master. I don't think we need more than a few lines of change for that.